manifests/init.pp

```diff
@@ -445,153 +445,130 @@
-  $api_user                       = undef,
-  $api_password                   = undef,
-  $api_ssl_port                   = undef,
-  $api_ssl_keystore_file          = undef,
-  $api_ssl_keystore_password      = undef,
-  $subscriptions                  = [],
-  $client_bind                    = '127.0.0.1',
-  $client_port                    = '3030',
-  $client_address                 = $::ipaddress,
-  $client_name                    = $::fqdn,
-  $client_custom                  = {},
-  $client_keepalive               = {},
-  $safe_mode                      = false,
-  $plugins                        = [],
-  $plugins_defaults               = {},
-  $plugins_dir                    = undef,
-  $purge                          = false,
-  $purge_config                   = false,
-  $purge_plugins_dir              = false,
-  $use_embedded_ruby              = true,
-  $rubyopt                        = undef,
-  $gem_path                       = undef,
-  $log_level                      = 'info',
-  $log_dir                        = '/var/log/sensu',
-  $dashboard                      = false,
-  $init_stop_max_wait             = 10,
-  $gem_install_options            = undef,
-  $hasrestart                     = true,
-  $enterprise_dashboard_base_path = undef,
-  $enterprise_dashboard_host      = undef,
-  $enterprise_dashboard_port      = undef,
-  $enterprise_dashboard_refresh   = undef,
-  $enterprise_dashboard_user      = undef,
-  $enterprise_dashboard_pass      = undef,
-  $enterprise_dashboard_ssl       = undef,
-  $enterprise_dashboard_audit     = undef,
-  $enterprise_dashboard_github    = undef,
-  $enterprise_dashboard_gitlab    = undef,
-  $enterprise_dashboard_ldap      = undef,
-  $path                           = '$PATH',
-  $redact                         = undef,
-  $deregister_on_stop             = false,
-  $deregister_handler             = undef,
-  $package_checksum               = undef,
-  $windows_pkg_url                = undef,
-  $windows_repo_prefix            = 'https://repositories.sensuapp.org/msi',
-  $windows_logrotate              = false,
-  $windows_log_number             = '10',
-  $windows_log_size               = '10240',
-  $windows_package_provider       = undef,
-  $windows_choco_repo             = undef,
-  $windows_package_name           = 'Sensu',
-  $windows_package_title          = 'sensu',
+  Pattern[/^absent$/, /^installed$/, /^latest$/, /^present$/, /^\d[\d\.\-\w]+$/] $sensu_plugin_version = 'installed',
+  Boolean            $install_repo = true,
+  Boolean            $enterprise = false,
+  Pattern[/^absent$/,/^installed$/,/^latest$/,/^present$/,/^[\d\.\-]+$/] $enterprise_version = 'latest',
+  Optional[String]   $enterprise_user = undef,
+  Optional[String]   $enterprise_pass = undef,
+  Boolean            $enterprise_dashboard = false,
+  String             $enterprise_dashboard_version = 'latest',
+  Boolean            $manage_repo = true,
+  Enum['main','unstable'] $repo = 'main',
```

Comment:
OK, I don’t have a strong preference either way since the manifest code is using a selector over this variable anyway.